Add new vignette (#68)
insightsengineering/teal.picks: vignettes/teal-picks-in-teal.Rmd

Reviewer: Please make a change in `_pkgdown.yml` to include all vignettes in site build

Author: Done

<!-- # nolint end: object_length_linter. -->

### Define choices and selections

Each `picks()` element has `choices` and `selected`. You can define them in four ways:

| Approach | Works with |
|---|---|

> | Default behavior | `variables()`, `values()`, and `datasets()` (`selected` only) |

Reviewer: This is not true to datasets, please remove the exception and maybe add a small description

```r
resolver(teal.picks::picks(datasets()), data = teal.data::teal_data(ADSL = rADSL, ADTTE = rADTTE))
#' <picks>
#'   <datasets>:
#'     choices: ADSL, ADTTE
#'     selected: ADSL
#'     multiple=FALSE, ordered=FALSE, fixed=FALSE
```

```suggestion
| Default behavior - no arguments needed for generic usage | `datasets()`, `variables()`, `values()`
```

| Static — character vector, integer index, or numeric range | `datasets()`, `variables()`, `values()` |
| `tidyselect` helpers — `everything()`, `starts_with()`, `where()`, … | `datasets()`, `variables()` only |
| Function — an R function applied at runtime to the data | `datasets()`, `variables()`, `values()` |

The sections below walk through each approach with runnable examples.

#### Defaults
Defaults depend on the slot type.

> For `datasets()`, there is no default for `choices`; define at least one dataset unless `check_no_dataset = FALSE`.

Reviewer: ```suggestion
For `datasets()`, the default `choices` are all available datasets. A `picks()` call needs a `datasets()` as first positional argument, except when `check_no_dataset = FALSE`.
```

> For `datasets()`, there is no default for `choices`; define at least one dataset unless `check_no_dataset = FALSE`.

Reviewer: Maybe change this to a bullet to make it more readable instead of a single line for all

Author: True, fix it [here](https://github.com/insightsengineering/teal.picks/pull/68/changes/80bfd5e71d909c683d27630ef21646d485af707c):

For `variables()`, the default `choices` is all variables in the selected dataset.
For `values()`, the default `choices` is all values of the selected variable.
For `selected`, the default is the first available choice, or all choices when `multiple = TRUE`.

```{r defaults}
picks(
  datasets(choices = "ADSL", selected = "ADSL"),
  variables()
)

picks(
  datasets(choices = "ADSL", selected = "ADSL"),
  variables(choices = "SEX", selected = "SEX", multiple = FALSE),
  values()
)
```

#### Static choices

Pass a character vector to `choices` to enumerate options exactly. Use `selected` to set the

> default. Integer indices work too (for example `selected = 1L` means the first element of

Reviewer: ```suggestion
Pass a character vector to `choices` to enumerate options exactly. Use `selected` to set the default.
Integer indices work too (for example `selected = 1L` means the first element of
```

`choices`).

```{r static}
# Datasets — user may switch between ADSL and ADLB; ADSL is the default
p_datasets <- picks(
  datasets(
    choices  = c("ADSL", "ADLB"),
    selected = "ADSL"
  )
)

# Variables — only a named subset is offered; first column pre-selected
p_variables <- picks(
  datasets(choices = "ADSL", selected = "ADSL"),
  variables(
    choices  = c("AGE", "SEX", "ARM"),
    selected = "AGE",
    multiple = FALSE
  )
)

# Values — categorical filter; two levels pre-selected
p_values <- picks(
  datasets(choices = "ADSL", selected = "ADSL"),
  variables(choices = "SEX", selected = "SEX", multiple = FALSE),
  values(
    choices  = c("M", "F"),
    selected = "F"
  )
)

p_datasets
p_variables
p_values
```

#### `tidyselect` helpers

`tidyselect` predicates let `choices` and `selected` adapt to the actual data at runtime instead

> of being hard-coded. This is supported for `datasets()` and `variables()`.

Reviewer: ```suggestion
`tidyselect` predicates let `choices` and `selected` adapt to the actual data at runtime instead of being hard-coded.
This is supported for `datasets()` and `variables()`.
```

Commonly used helpers:

- `tidyselect::everything()` — all items
- `tidyselect::starts_with()` / `tidyselect::ends_with()` / `tidyselect::contains()` — pattern matching
- `tidyselect::matches()` — regex matching
- `tidyselect::where(predicate)` — columns satisfying a predicate function
- `tidyselect::all_of()` / `tidyselect::any_of()` — vector-based selection (error-safe or silent)
- Integer indices such as `1L`, `1L:3L` — select by position

> Note: `tidyselect` is not supported by `values()`. Use explicit vectors or a function

> > there instead (see [Functions](#functions)).

Reviewer: ```suggestion
> Note: `tidyselect` is not supported by `values()`.
> Use explicit vectors or a function there instead (see [Functions](#functions)).
```

```{r tidyselect}
# Datasets — offer any data.frame in the teal_data object
p_any_dataset <- picks(
  datasets(
    choices  = tidyselect::where(is.data.frame),
    selected = 1L # first dataset by default
  )
)

# Variables — all numeric columns; first one pre-selected
p_numeric_vars <- picks(
  datasets(choices = "ADSL", selected = "ADSL"),
  variables(
    choices  = tidyselect::where(is.numeric),
    selected = 1L,
    multiple = FALSE
  )
)

# Variables — columns whose names start with "A"; first two pre-selected
p_a_prefix <- picks(
  datasets(choices = "ADSL", selected = "ADSL"),
  variables(
    choices  = tidyselect::starts_with("A"),
    selected = 1L:2L,
    multiple = TRUE
  )
)

p_any_dataset
p_numeric_vars
p_a_prefix
```

#### Functions {#functions}

You can pass a plain R function as `choices` or `selected`. The function receives the relevant
context object (the current dataset for `datasets()` and `variables()`, the current column vector
for `values()`) and must return the subset to use. This is the only runtime-dynamic approach

> supported by `values()`.

Reviewer: ```suggestion
for `values()`) and must return the subset to use.
This is the only runtime-dynamic approach supported by `values()`.
```
